dipy/reconst/csdeconv.py

```diff
@@ -118,18 +120,20 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
         # scale lambda_ to account for differences in the number of
         # SH coefficients and number of mapped directions
         # This is exactly what is done in [4]_
-        self.lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
-                        (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.))
-                       )
+
+        lambda_ = (lambda_  * self.R.shape[0] * r_rh[0] /
+                   (np.sqrt(self.B_reg.shape[0]) * np.sqrt(362.)))
```

Comment:
@samuelstjean I'm not sure what you would fix, here is the historical accident:

The CSD equation is:

 $F(f_n) = ||Xf_n - S||^2 + \lambda^2 ||H_{n-1} f_n||^2$

Turns out this equation will give a different result if you use a more dense sphere (sphere with more points), unless you also change $\lambda$. So we can reformulate the the equation as:

 $F(f_n) = ||Xf_n - S||^2 + (\lambda')^2 / N * ||H_{n-1} f_n||^2$

Now the result is not sensitive to the number of points on the sphere and the $\lambda'$ can be used for all spheres (but the result is still sensitive to how well the points are distributed on the sphere). To make sure that dipy gives reasonable result by default we need to set $\lambda'_{default} \approx \lambda_{default} * 20$. This method would work for code which uses all defaults, but existing user code which set $\lambda$ explicitly would give different results before and after the change. Instead we can write the equation as:

 $F(f_n) = ||Xf_n - S||^2 + (\lambda')^2 * N_0 / N  *||H_{n-1} f_n||^2$

With this formulation when $N = N_0$ $\lambda'$ can simply be set to $\lambda$ and dipy will give the same result. Now the question is how to pick $N_0$. We picked $N_0 = 362$ because that's the number of points on the default sphere, which we assume is the sphere that is most often used by our users.

I'm not sure there is anything to fix here, except maybe adding this blurb as a comment so that I can avoid this conversation in the future :).